teps/0076-array-result-types.md

```diff
@@ -452,8 +452,27 @@ tasks:
 Additionally, array results can be passed directly to Tasks which take arrays as parameters using
 [the `[*]` variable replacement syntax](https://github.com/tektoncd/pipeline/blob/main/docs/tasks.md#substituting-array-parameters).
 
-For example:
-
+For example, when params `update-all-environments` in array type is substituted with another array type `tasks.get-environments.results.environments`:
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+...
+spec:
+tasks:
+  - name: update-all-environments
+    params:
+      - name: environments
+        value: $(tasks.get-environments.results.environments[*])
```

Comment:
I think the first two examples are overlapping. We don't need to distinguish whether the variable reference is quoted or not. It's just K8s default behaviour to treat any value as string if it's not array or dictionary no matter if it is quoted ([doc](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/)). i.e.
```yaml
params:
      - name: environments
        value: abc
```
is equivalent to 
```yaml
params:
      - name: environments
        value: "abc"
```
```

Reply:
> I think the first two examples are overlapping. We don't need to distinguish whether the variable reference is quoted or not. It's just K8s default behaviour to treat any value as string if it's not array or dictionary no matter if it is quoted ([doc](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/)).

I believe that the yaml and json conversions are through K8S API (OpenAPI) and it is generated through [swagger.json](https://github.com/tektoncd/pipeline/blob/main/pkg/apis/pipeline/v1beta1/swagger.json)
And the value field under params is marshaled and unmarshaled by the implemented `json.Unmarshaller interface` in params_types.go

Reply:
Is there a functional difference between how `value: foo` and `value: "foo"` is interpreted though? (e.g. they would come in as a string, but we'd recognize the [*] syntax and treat it like an array)